[LLVM][OpenMPOpt] Fix a crash when associated function is nullptr

shiltian opened the pull request:
The associated function can be a nullptr if it is an indirect call.
This causes a crash in `CheckCallee` which always assumes the callee
is a valid pointer.

Fix #66904.

jplehr commented on llvm/lib/Transforms/IPO/OpenMPOpt.cpp:
```diff
@@ -5028,7 +5028,9 @@ struct AAKernelInfoCallSite : AAKernelInfo {
     if (!AACE || !AACE->getState().isValidState() || AACE->hasUnknownCallee()) {
-      CheckCallee(getAssociatedFunction(), /*NumCallees=*/1);
+      Function *F = getAssociatedFunction();
+      if (F)
```
Is there a specific reason to not reuse F in the call to `ChackCallee`?

```
if (Function *F = getAssociatedFunction())
  CheckCallee(F, 1);
```

jplehr reviewed (commented):
LG

shiltian committed:
[LLVM][OpenMPOpt] Fix a crash when associated function is nullptr

The associated function can be a nullptr if it is an indirect call.
This causes a crash in `CheckCallee` which always assumes the callee
is a valid pointer.

Fix #66904.

shiltian committed:
fix comment

jplehr reviewed (approved):
LGTM

jdoerfert commented on llvm/lib/Transforms/IPO/OpenMPOpt.cpp:
```diff
@@ -5028,7 +5028,9 @@ struct AAKernelInfoCallSite : AAKernelInfo {
-      CheckCallee(getAssociatedFunction(), /*NumCallees=*/1);
+      Function *F = getAssociatedFunction();
+      if (F)
+        CheckCallee(getAssociatedFunction(), /*NumCallees=*/1);
```
F
